athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/S3BlockSpiller.java

```diff
@@ -512,7 +514,7 @@ private ThreadPoolExecutor makeAsyncSpillPool(SpillConfig config)
                 }
                 catch (InterruptedException e) {
                     Thread.currentThread().interrupt();
-                    throw new RejectedExecutionException("Received an exception while submitting spillBlock task: ", e);
+                    throw new AthenaConnectorException("Received an exception while submitting spillBlock task: ", ErrorDetails.builder().errorCode(FederationSourceErrorCode.INTERNAL_SERVICE_EXCEPTION.toString()).build());
```

Comment:
This needs to stay as a RejectedExecutionException. The caller can apply a retry policy and the retry policy expects a RejectedExecutionException.

Reply:
reverted back.